### PR TITLE
Update/actions

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** GitHub Actions need to be updated.

**Description of the Change:** GH detected a [high-severity](https://github.com/advisories/GHSA-cxww-7g56-2vh6) vulnerability, with the download artifact action.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-72698]

### CODE REVIEW

File: .github/workflows/post_release_version_bump.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
           --changelog_path "./.github/CHANGELOG.md" --post_release
 
       - name: Create Pull Request
-        uses: peter-evans/create-pull-request@v3
+        uses: peter-evans/create-pull-request@v7
```
Comment (vincentmr) [inline .github/workflows/post_release_version_bump.yml:32]: Are we sure this won't break the workflow?

File: .github/workflows/tests_lgpu_python.yml [hunk 2]
```diff
@@ -235,11 +235,13 @@ jobs:
           mv coverage.xml coverage-${{ github.job }}-${{ matrix.pl_backend }}.xml
 
       - name: Upload code coverage results
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: ubuntu-codecov-results-python
           path: ./main/coverage-${{ github.job }}-${{ matrix.pl_backend }}.xml
```
Comment (vincentmr) [inline .github/workflows/tests_lgpu_python.yml:241]: Is there no conflict between backends here?

**Review by vincentmr — COMMENTED** [review]
Thanks @AmintorDusko , disk space is a real nuisance so I hope `retention-days: 1` will help.

**Review by maliasadi — APPROVED** [review]
Nice work! Thanks @AmintorDusko :partying_face: Feel free to merge it after updating the changelog.

File: .github/workflows/post_release_version_bump.yml (hunk 1 above)
Comment (AmintorDusko) [reply .github/workflows/post_release_version_bump.yml:32]: Sure, Sure we don't. I'm optimistic, though. I can undo the change if you think it is too risky.

File: .github/workflows/tests_lgpu_python.yml (hunk 2 above)
Comment (AmintorDusko) [reply .github/workflows/tests_lgpu_python.yml:241]: We have a single Backend here. I will expand the name, though.

File: .github/workflows/post_release_version_bump.yml (hunk 1 above)
Comment (vincentmr) [reply .github/workflows/post_release_version_bump.yml:32]: I'm ok either way. Let's see how it does next release.

**Review by vincentmr — APPROVED** [review]
LGTM, thanks again @AmintorDusko .

**Pull request merged**